### ISSUE TITLE
Feature: Printing via Screen shotting

### ISSUE THREAD

**Issue by gedw99** [go-flutter-desktop/go-flutter#63]
This looks like an excellent basis.

https://github.com/DavBfr/dart_pdf
It can take a screenshot and then output as as PDF and then Share / Print.

Taking a screenshot is a bit hacky to purists, but way let work than converting Dart AST layout to PDF.
SO its not a bad solution in terms of easy but effective in 90% of cases approach i feel.

In progres is the ability to take a PNG of a Flutter VIew that is larger than the page. For example when you have a large scroll list.
See: https://github.com/DavBfr/dart_pdf/issues/28

I think the plugin ( java, swift) code can be replaced with pure golang stuff reasonably easily.
The Dart code looks very promising. I would like to collaborate rather than fork, because it would be very useful to the community to have plugins that work on mobile and Desktop. Now that we have the downloader it starts to be pretty easy i think.

**Comment by GeertJohan** [comment]
Hi @gedw99, sounds very interesting. I've been thinking about possibilities to supplement existing plugins with desktop functionality. Have you tried this already or thought about such a solution?

**Comment by gedw99** [comment]
Hey @GeertJohan 

BTW i saw the issue about the new org and you taking over maintenance. Huge thank you.
I hope i can help also in my own way.

Regarding your question "about possibilities to supplement existing plugins with desktop functionality", i am not exactly sure what you mean. Can you be more verbose ?

**Comment by GeertJohan** [comment]
Thanks @gedw99 

There are a lot of plugins already out there, I'm thinking if and how it's possible to add desktop (Go-based) support for those plugins. So that the dart side stays the same, but the native implementation is swapped for a Go version. I think this must be possible with the messaging system.

**Comment by gedw99** [comment]
I think it's very possible and have seen 2 examples.

https://github.com/empirefox/flutter_dial_go

https://sites.google.com/a/athaydes.com/renato-athaydes/posts/buildingamobilefrontendforagoapplicationusingflutter

I have tried both. The 2nd is closest because it's just JSON RPC.very clean. It's designed for gomobile obviously, but a slightly different driver wrapper can be built I suspect to match the desktop API. But I have not tried or put that much though into it 

For desktop I went with the golang code running as a service, and the flutter GUI in a different process. The golang service uses nomad so that many golang processes can be run. So the plugins are a really side cars.
For example a print server. Or just another desktop side Microservices.
It's still very much an idea with bits of code. Nothing stable yet architecturally.
The core requirement for me was to add desktop side Microservices at runtime without recompiling. Then I can do the exact same thing server side and not use kubernettes 
NATS is being experimented with for joining them up and it's loaded in the core service.


But I do agree that a pluginodel for things that are GUI related like. Dialogs, etc is needed.
